pkg/flowaggregator/flowaggregator.go

```diff
@@ -248,7 +268,10 @@ func (fag *flowAggregator) sendTemplateRecord(templateSet ipfix.IPFIXSet) (int,
 		ie := ipfixentities.NewInfoElementWithValue(element, nil)
 		elements = append(elements, ie)
```

Comment:
do you need elements now? once the sanity check passes you can use `fag.infoElementsOrder` directly.

Reply:
`fag.infoElementsOrder` is only a list of element string, we need to build elements which is a InfoElementWithValue list to send out template record.

Reply:
Yes, but as we discussed in this comment https://github.com/vmware-tanzu/antrea/pull/1596#discussion_r532774979, the elements order returned by GetOrderedElementList() is not as I expect.

Reply:
Could you change your order to match the order in which the intermediate process stores the elements in its record? Is that not possible? 
I believe all records have the same order. Please correct me if that is not the case. 
By reusing the elements, we can optimize the memory allocations.

Reply:
Yes it's possible and I could do that, the reason I didn't do it because I think the order returned by GetOrderedElementList() is not appropriate. I remembered "originalExporterIPv4Address" and "originalObservationDomainId" ordered behind all AntreaInfoElements in GetOrderedElementList(), but these two are actually IANAInfoElements, so I choose to define a new order list for flow aggregator.

Reply:
If I understand correctly, you want to place all IANA infoelements together and antrea infoelements together. Is there any specific reason? Order does not matter for the collector, which is the characteristic of the protocol.  Whatever order we send the template, collector expects in the same order. Therefore, if we modify our template to reflect the order given by intermediate process, then we are good.

Reply:
Yes, it's true that we can modify the template to reflect the order given by intermediate process manually, but the code in sendTemplateRecord() will become something like this which I think it's not elegant:
````
for _, ie := range IANAInfoElements except "originalExporterIPv4Address" and "originalObservationDomainId":
    do something...
for _, ie := range IANAReverseInfoElements:
    do something...
for _, ie := range "originalExporterIPv4Address" and "originalObservationDomainId":
    do something...
for _, ie := range AntreaInfoElements:
    do something...

Reply:
Just create another list for "originalExporterIPv4Address" and "originalObservationDomainId" with a different name such as aggregatorElements and the other one as exporterElements.

Prefer multiple lists to unnecessary allocations, which can be avoided to decrease the overhead.

Btw, are those lists need to be public?

Reply:
Resolved conversation offline.